rpc/handlers.go

```diff
@@ -1036,12 +1039,17 @@ func (h *Handler) TransactionStatus(ctx context.Context, hash felt.Felt) (*Trans
 			Finality:  receipt.FinalityStatus,
 			Execution: receipt.ExecutionStatus,
 		}
+		h.txStatusCache.Delete(hash)
 	case ErrTxnHashNotFound:
 		txStatus, err := h.feederClient.Transaction(ctx, &hash)
```

Comment:
I guess it's overkill to create cache over `feederClient.Transaction(...)` so any caller would benefit from the cache.